aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.kt

```diff
@@ -235,21 +255,33 @@ internal class AWSS3StorageService(
                     null
                 }
             }
-            StorageListResult.fromItems(items, result.nextContinuationToken)
+
+            // remove prefixes in Gen1 code path (i.e. remove "public", "private", "protected" prefixes
```

Comment:
Curious about this - was this covered in the design review? Will it cause issues for existing Gen1 customers?

Reply:
Good call out, I was trying to make the implementation consistent with [Amplify Swift's implementation](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService%2BListBehavior.swift#L50).  But it looks like on Android, paths are always returned as full path with the prefixes still intact.  I've updated to keep the prefixes in Android to be consistent.